src/SimpleEventStore/SimpleEventStore.CosmosDb/AzureCosmosDbStorageEngine.cs

```diff
@@ -88,25 +88,39 @@ public async Task<IStorageEngine> Initialise(CancellationToken cancellationToken
                 ? int.MaxValue
                 : startPosition + numberOfEventsToRead;
 
-            var eventsQuery = _collection.GetItemLinqQueryable<CosmosDbStorageEvent>()
-                .Where(x => x.StreamId == streamId && x.EventNumber >= startPosition && x.EventNumber <= endPosition)
-                .OrderBy(x => x.EventNumber)
-                .ToFeedIterator();
-
-            var events = new List<StorageEvent>();
-
-            while (eventsQuery.HasMoreResults)
+            var queryDefinition = new QueryDefinition(@"
+                    SELECT VALUE e
+                    FROM e
+                    WHERE e.streamId = @StreamId
```

Comment:
There was an interesting article on the newer library having a StringEquals which lowered RUs. Might be worth trying that? 
https://devblogs.microsoft.com/cosmosdb/new-string-function-performance-improvements-and-case-insensitive-search/

Reply:
Unless streamId is an int actually

Reply:
Not sure that matters here as we're doing a case sensitive exact match.